packages/salesforce-adapter/src/adapter.ts

```diff
@@ -811,6 +812,15 @@ export default class SalesforceAdapter implements SalesforceAdapterOperations {
     // Check old configuration flag for backwards compatibility (SALTO-2700)
     const checkOnly = this.userConfig?.client?.deploy?.checkOnly ?? false
     const result = await this.deployOrValidate(deployOptions, checkOnly)
+    const enrichedErrors = await enrichSaltoDeployErrors(
+      result.errors,
+      deployOptions.getAllElements ?? (async () => []),
+    )
+    const friendlyErrors = enrichedErrors.map(error => ({
+      ...error,
+      detailedMessage: getUserFriendlyDeployErrorMessage(error),
+    }))
+    result.errors = friendlyErrors
```

Comment:
I prefer not defining consts that are only used once unless it really affects readability.
```suggestion
    result.errors = (await enrichSaltoDeployErrors(
      result.errors,
      deployOptions.getAllElements ?? (async () => []),
    )).map(error => ({
      ...error,
      detailedMessage: getUserFriendlyDeployErrorMessage(error),
    }))
```